articles/zk-snarks-under-the-hood/zk_snarks_under_the_hood.md

```diff
@@ -0,0 +1,121 @@
+---
+Title: "Zk-SNARKs: Under the Hood"
+Status: Done
+Level: 6
+Note: There is a few places that are hard to fully understand
+---
+
+# Zk-SNARKs: Under the Hood
+
+[Site](https://medium.com/@VitalikButerin/zk-snarks-under-the-hood-b33151a013f6)
+
+[Pinocchio protocol](https://eprint.iacr.org/2013/279.pdf) Parno, Gentry, Howell and Raykova from 2013 (often called
+PGHR13)
+
+Supposed that a pair of points $(P, Q)$ falls from the sky, where $P * k = Q$, but nobody knows what the value of $k$
+is. Now, suppose that I come up with a pair of points $(R, S)$ where $R * k = S$. Then,
+the [knowledge of exponent](../../terms/knowledge_of_exponent.md) assumption implies that the only way I could have made
+that pair of points was by taking $P$ and $Q$, and multiplying both by some factor r_that I personally know_.
+
+Checking $R * k = S$_doesn’t actually require knowing k -_instead, you can simply check whether or not $e(R, Q) = e(P,
```

Comment:
Check this math expression

Reply:
this is not the file related to this Pull request, so we shouldn't care about this, the PR for this is SOT 39